cmd/antrea-agent/options.go

```diff
@@ -273,3 +276,18 @@ func (o *Options) validateFlowExporterConfig() error {
 	}
 	return nil
 }
+
+func (o *Options) validateFlexibleIPAMConfig() error {
+	if features.DefaultFeatureGate.Enabled(features.FlexibleIPAM) {
+		if o.config.OVSDatapathType != string(ovsconfig.OVSDatapathSystem) {
+			return fmt.Errorf("FlexibleIPAM requires 'system' OVSDatapathType")
+		}
+		if o.config.TrafficEncapMode != config.TrafficEncapModeNoEncap.String() {
+			return fmt.Errorf("FlexibleIPAM requires 'NoEncap' TrafficEncapMode")
+		}
+		if !o.config.NoSNAT {
```

Comment:
Why we need noSNAT?

Reply:
In the current design, the L2 Gateway will use static route for per-Node IPAM Pods' traffic and no SNAT needed.
We can remove this restriction, but it will be SNATed for per-Node IPAM Pod -> Antrea-IPAM Pod traffic since Antrea-IPAM IPs on remote Nodes are not recognized on local Node.

Reply:
Could not we add iptables rule to bypass all IPPools?

Another question - could we support Node IPAM with encap, and flexible IPAM with noEncap? If not what is the major issue to support it?

Reply:
Yes, but this method will introduce scale issue because we need to add all the PodIPs in IPPools to iptables, and to keep it updated when IPPool modified.
For the second question, this can be supported by adding some flows, but this config may confuse user. Do you think we need to support this?

Reply:
I feel it might be ok to add subnets in IPPools, assuming in most cases we will have subnets per Namespace. But I am fine if you want to support only noSNAT first, but please add some comments to explain the reason.

Reply:
Sure. Added TODO here.

Reply:
Could it be ok to assume that we don't want to SNAT any IP which is not in the CIDR (or CIDRs) specified for NodeIpam?

Reply:
@salv-orlando I think this is already implemented and not the issue discussed in this thread.
See https://github.com/antrea-io/antrea/blob/main/pkg/agent/route/route_linux.go#L412

The current issue is this iptables rule will SNAT the traffic from Per-Node IPAM Pod to AntreaIPAM Pod, because AntreaIPAM IPs are not recognized in this iptables chain and will be SNATed by above rule.